laravel/horizon
# feat(i18n): support i18n for horizon

## What does this Pull Request do

Added internationalization support for Laravel Horizon based on VueI18n, currently supports English and Chinese languages.

## Change details

### Added

- Added `vue-i18n` and `date-fns` dependencies
- Added  two new files `en.json` and `zh.json` to the `resources/lang` directory, containing all the text strings that need to be translated in the Horizon interface.
- Added the initialization code for the VueI18n plugin to `resources/js/app.js`, which loads Horizon's translation files and binds them to the Vue instance. The default language is set to `en`.
- Added i18n toggle button to `resources/views/layout.blade.php` 

### Modified
- Adapted the strings in the Blade template to Vue plugins for easy toggling through the interface.
- Modify the components/views under `resources/screens/` and `resources/components/`, use the `$t` method to mark the strings that need to be translated.

## Why these changes are necessary

Laravel Horizon is a very popular queue management tool that is widely used in various PHP applications. Due to its open-source nature, many international users also want to use this tool, but currently, the interface text of Horizon is fixed in English, which is not very user-friendly for non-English users.

To solve this problem, this PR adds internationalization support to Horizon based on Vue.js, making it easy for more people to use this tool.

## How to test

- Install Horizon in a Laravel example-app.
- Use this Pull Request branch to install Horizon or replace the contents under `vendor/laravel/horizon`, use `mix` to compile and use `php artisan horizon:publish` to publish assets.
- Visit the Horizon Dashboard and try switching between different language to see if the text on the interface can be switched correctly.
